init.lua

```diff
@@ -540,7 +528,7 @@ require('lazy').setup({
           -- code, if the language server you are using supports them
           --
           -- This may be unwanted, since they displace some of your code
-          if client and client.server_capabilities.inlayHintProvider and vim.lsp.inlay_hint then
+          if client and client.supports_method(vim.lsp.protocol.Methods.textDocument_inlayHint) then
             map('<leader>th', function()
               vim.lsp.inlay_hint.enable(not vim.lsp.inlay_hint.is_enabled())
```

Comment:
LSP is warning the following: `This function requires 1 argument(s) but instead it is receiving 0.`

```suggestion
              vim.lsp.inlay_hint.enable(not vim.lsp.inlay_hint.is_enabled {})
```

Reply:
Thanks, that works.

Reply:
YMMV, but I had issues until I passed the buffer in the filter table:

```suggestion
              vim.lsp.inlay_hint.enable(not vim.lsp.inlay_hint.is_enabled({ bufnr = event.buf }))
```